### PR TITLE
add kafka config property

### PR DESCRIPTION
**Pull request by nharring-adjacent**
This change allows for setting `fetch.message.max.bytes` from a CREATE SOURCE WITH options, such as:
```sql
CREATE SOURCE local_kafka
FROM KAFKA BROKER 'localhost:9092' TOPIC 'data'
WITH (fetch_message_max_bytes=1000)
FORMAT bytes;
```

To test I added logging at the point of rdkafka client creation and the logs for the example query confirm the parameter being passed through with the correct value:
```
2021-12-16T18:34:01.954491Z  INFO coord::catalog: create view materialize.public.json_test_source (u10)
2021-12-16T18:34:01.954574Z  INFO coord::catalog: create index materialize.public.json_test_source_primary_idx (u11)
2021-12-16T18:34:01.957067Z  INFO dataflow::source::kafka: client has config: ClientConfig { conf_map: {"statistics.interval.ms": "1000", "group.id": "materialize-47df596e-1cdd-4a6f-864d-ac629bca43e7-kafka-u9/31", "auto.offset.reset": "earliest", "isolation.level": "read_committed", "bootstrap.servers": "localhost:9092", "fetch.message.max.bytes": "1", "enable.auto.commit": "false", "topic.metadata.refresh.interval.ms": "30000"}, log_level: Info }
```
### Motivation

  * This PR adds a known-desirable feature: [#6686](https://github.com/MaterializeInc/materialize/issues/6686)

### Checklist

- [X] This PR has adequate test coverage / QA involvement has been duly considered.
- [X] This PR adds a release note for any [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/user/content/release-notes.md#what-changes-require-a-release-note).

### CODE REVIEW

**Comment by delvinshen-optum** [comment]
I have a question on log4j CVE-2021-44228 security vulnerability issue on Materialize product:
A 0-day exploit (CVE-2021-44228 – also called log4shell) was discovered in the popular Java logging library log4j2 that allows remote code execution (RCE) by logging a specific string (similar to SQL injections):

The vulnerability occurs in the log library log4j version >= 2.0-beta9 and < 2.15 . The older version of the log4j library version 1 is not directly affected as of today.

Could you please verify this Log4J security vulnerability issue with your products, and provide users your verification results.

**Review by petrosagg — APPROVED** [review]
Nice! I didn't know adding a kafka option is that simple :) LGTM, do you mind adding an entry in our docs for this option?

File: doc/user/layouts/partials/create-source/connector/kafka/with-options.html [hunk 1]
```diff
@@ -9,6 +9,7 @@
 `timestamp_frequency_ms`| `int` | Default: `1000`. Sets the timestamping frequency in `ms`. Reflects how frequently the source advances its timestamp. This measure reflects how stale data in views will be. Lower values result in more-up-to-date views but may reduce throughput.
 `topic_metadata_refresh_interval_ms` | `int` | Default: `30000`. Sets the frequency in `ms` at which the system checks for new partitions. Accepts values [0,3600000].
 `enable_auto_commit` | `boolean`| Default: `false`. Controls whether or not Materialize commits read offsets back into Kafka. This is purely for consumer progress monitoring and does not cause Materialize to resume reading from where it left off across restarts.
+`fetch_message_max_bytes` | `int` | Default: `134217728`.  Accepts values [1, 1000000000], controls the initial maximum number of bytes per topic+partition to request when fetching messages from the broker. If the client encounters a message larger than this value it will gradually try to increase it until the entire message can be fetched.
```
Comment (sploiselle) [inline doc/user/layouts/partials/create-source/connector/kafka/with-options.html:12]: Can you pattern this after one of the other lines here? e.g.

> `topic_metadata_refresh_interval_ms` | `int` | Default: `30000`. Sets the frequency in `ms` at which the system checks for new partitions. Accepts values [0,3600000].

Comment (nharring-adjacent) [reply doc/user/layouts/partials/create-source/connector/kafka/with-options.html]: I moved the other one which was mid-line to the end of the line as well so all three are formatted the same way.

**Pull request merged**